src/AutoChessSharp.Core/Model/Store.cs

```diff
@@ -5,28 +5,28 @@
 /// </summary>
 public class Store
 {
-    private List<Piece>? _storePieces;
-    private List<Piece> _availPieces;
+    private List<IPiece>? _storePieces;
+    private IEnumerable<IPiece> _availPieces;
```

Comment:
IEnumerable of IPieces to allow piece database read from all kinds of collection primitives, the internal store pieces configured to list of pieces